### PR TITLE
Release prep v0.6.0

### PR DESCRIPTION
**Pull request by ircwaves**
**Related Issue(s):**

- #133 

**PR Checklist:**

- [X] I have added my changes to the [CHANGELOG](https://github.com/stac-utils/stac-server/blob/main/CHANGELOG.md) **or** a CHANGELOG entry is not required.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [project]
 name = "stactask"
-version = "0.5.1"
+version = "0.6.0"
 authors = [{ name = "Matthew Hanson", email = "matt.a.hanson@gmail.com" }]
 maintainers = [{ name = "Pete Gadomski", email = "pete.gadomski@gmail.com" }]
```
Comment (gadomski) [inline pyproject.toml:5]: @ircwaves can you remove me as a maintainer, and maybe add yourself or someone else? 🙇🏼

**Review by gadomski — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (gadomski) [reply pyproject.toml:5]: To be clear, still happy to review stuff! I just don't feel like I'm in the position to be The Maintainer™ anymore :-P

**Comment by ircwaves** [comment]
FWIW, I think that the move to instance method here presents minimal impact to any existing implementations which upgrade to this release.  @gadomski do you think this merits another `Migration` section like this:

> ### 0.5.x -> 0.6.0
> 
> Previously, the `validate` method was a _classmethod_, validating the payload
> argument passed.  This has now been made an instance method, which validates
> the `self._payload` copy of the payload which the `Task` instance during
> construction.  This is behaviorally the same, in that construction will fail.
> 
> Previous implementations of `validate` would have been similar to this:
> 
> ```python
>     @classmethod
>     def validate(payload: dict[str, Any]) -> bool:
>         # Check The Things™
>         return isinstance(payload, dict)
> ```
> 
> And will now need to be updated to this form:
> 
> ```python
>     def validate(self) -> bool:
>         # Check The Things™
>         return isinstance(self._payload, dict)
> ```
>

**Comment by gadomski** [comment]
> @gadomski do you think this merits another Migration section like this:

Can't hurt, right? :-)

File: pyproject.toml [hunk 2]
```diff
@@ -1,8 +1,11 @@
 [project]
 name = "stactask"
 version = "0.6.0"
-authors = [{ name = "Matthew Hanson", email = "matt.a.hanson@gmail.com" }]
-maintainers = [{ name = "Pete Gadomski", email = "pete.gadomski@gmail.com" }]
+authors = [
+  { name = "Matthew Hanson", email = "matt.a.hanson@gmail.com" },
+  { name = "Pete Gadomski", email = "pete.gadomski@gmail.com" },
```
Comment (gadomski) [inline pyproject.toml:6]: I don't feel like I'm an author of this package so can you remove me here? TIA.

**Review by gadomski — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 2 above)
Comment (ircwaves) [reply pyproject.toml]: Gotcha.

**Review by gadomski — APPROVED** [review]

**Pull request merged**